test/logger_json/formatters/elastic_test.exs

```diff
@@ -0,0 +1,415 @@
+      otel_span_id: ~c"bff20904aa5883a6",
+      otel_trace_flags: ~c"01",
+      otel_trace_id: ~c"294740ce41cc9f202dedb563db123532"
+    )
+
+    log =
+      capture_log(fn ->
+        Logger.debug("Hello")
+      end)
+      |> decode_or_print_error()
+
+    assert log["span.id"] == "bff20904aa5883a6"
+    assert log["trace.id"] == "294740ce41cc9f202dedb563db123532"
+  end
+
+  test "logs span and trace ids" do
+    Logger.metadata(
+      span_id: "bff20904aa5883a6",
+      trace_id: "294740ce41cc9f202dedb563db123532"
+    )
+
+    log =
+      capture_log(fn ->
+        Logger.debug("Hello")
+      end)
+      |> decode_or_print_error()
+
+    assert log["span.id"] == "bff20904aa5883a6"
+    assert log["trace.id"] == "294740ce41cc9f202dedb563db123532"
+  end
+
+  test "does not crash on invalid span and trace ids" do
+    Logger.metadata(
+      span_id: :foo,
+      trace_id: 123
+    )
+
+    log =
+      capture_log(fn ->
+        Logger.debug("Hello")
+      end)
+      |> decode_or_print_error()
+
+    assert log["span.id"] == "foo"
+    assert log["trace.id"] == 123
+  end
+
+  test "does not crash on invalid OTEL span and trace ids" do
+    Logger.metadata(
+      otel_span_id: :foo,
+      otel_trace_id: 123
+    )
+
+    log =
+      capture_log(fn ->
+        Logger.debug("Hello")
+      end)
+      |> decode_or_print_error()
+
+    assert log["span.id"] == "foo"
+    assert log["trace.id"] == 123
+  end
+
+  test "logs metadata" do
+    Logger.metadata(
+      date: Date.utc_today(),
+      time: Time.new(10, 10, 11),
+      pid: self(),
+      ref: make_ref(),
+      atom: :atom,
+      list: [1, 2, 3],
+      map: %{foo: :bar},
+      struct: URI.parse("https://example.com"),
+      binary: "binary",
+      node: node()
+    )
+
+    log_entry =
+      capture_log(fn ->
+        Logger.debug("Hello", float: 3.14)
+      end)
+      |> decode_or_print_error()
+
+    assert %{
+             "message" => "Hello",
+             "atom" => "atom",
+             "binary" => "binary",
+             "date" => _,
+             "list" => [1, 2, 3],
+             "map" => %{"foo" => "bar"},
+             "node" => "nonode@nohost",
+             "ref" => _ref,
+             "float" => 3.14,
+             "struct" => %{
+               "authority" => "example.com",
+               "fragment" => nil,
+               "host" => "example.com",
+               "path" => nil,
+               "port" => 443,
+               "query" => nil,
+               "scheme" => "https",
+               "userinfo" => nil
+             }
+           } = log_entry
+  end
+
+  test "logs exceptions" do
+    log_entry =
+      capture_log(fn ->
+        pid =
+          spawn(fn ->
+            raise RuntimeError
+          end)
+
+        ref = Process.monitor(pid)
+        assert_receive {:DOWN, ^ref, _, _, _}
+        Process.sleep(100)
+      end)
+      |> decode_or_print_error()
+
+    Jason.encode_to_iodata!(log_entry, pretty: true) |> IO.puts()
```

Comment:
@bvobart I think this one is left out by accident

Reply:
Oops, yes, you're right!